include/seqan3/range/view/translation.hpp

```diff
@@ -0,0 +1,330 @@
+    }
+};
+
+} // namespace seqan3::detail
+
+namespace seqan3::view
+{
+/*!\brief A view that translates nucleotide into aminoacid alphabet with 1, 2, 3 or 6 frames.
+ * \tparam irng_t The type of the range being processed.
+ * \param irange The range being processed.
+ * \param tf Translation frames (SINGLE_FRAME, WITH_REVERSE_COMPLEMENT, WITH_FRAME_SHIFTS, SIX_FRAME).
+ * \returns A range of ranges containing frames with aminoacid sequence. See below for the properties of the returned range.
+ * \ingroup view
+ *
+ * \details
+ *
+ * This view can be used to translate nucleotide sequences into aminoacid sequences:
+ * translation_frames::SINGLE_FRAME             - forward frame (Note: The output will be a range of ranges. If a single range
+ *                                                is wanted or needed, please use view::translate_single)
+ * translation_frames::WITH_REVERSE_COMPLEMENT  - forward/ reverse frames
+ * translation_frames::WITH_FRAME_SHIFTS        - all forward frames
+ * translation_frames::SIX_FRAME                - all forward and reverse frames
+ *
+ * \par View properties
+ *
+ * |                     | `irng_t` (range input type)           | `rrng_t` (range return type)                                |
+ * |---------------------|---------------------------------------|-------------------------------------------------------------|
+ * | range               | seqan3::forward_range_concept         | seqan3::view_concept && seqan3::random_access_range_concept |
+ * | `range_reference_t` | seqan3::nucleotide_concept            |                                                             |
+ *
+ * * `irng_t` is the type of the range modified by this view (input).
+ * * `rrng_type` is the type of the range returned by this view.
+ * * for more details, see \ref view.
+ *
+ * \par Example
+ *
+ * Operating on a range of seqan3::dna5:
+ * ```cpp
+ * dna5_vector vec{"ACGTACGTACGTA"_dna5};
+ *
+ * // single frame translation
+ * auto v1 = vec | view::translate_frames(SINGLE_FRAME);                                              // == [[T,Y,V,R]]
+ *
+ * // reverse translation
+ * auto v2 = vec | view::translate_frames(WITH_REVERSE_COMPLEMENT);                         // == [[T,Y,V,R],[Y,V,R,T]]
+ *
+ * // forward frames translation
+ * auto v3 = vec | view::translate_frames(WITH_FRAME_SHIFTS);                       // == [[T,Y,V,R],[R,T,Y,V],[V,R,T]]
+ *
+ * // six frame translation
+ * auto v4 = vec | view::translate_frames(SIX_FRAME);   // == [[T,Y,V,R],[R,T,Y,V],[V,R,T],[Y,V,R,T],[T,Y,V,R],[R,T,Y]]
+
+ * // function syntax
+ * auto v5 = view::translate_frames(vec, WITH_REVERSE_COMPLEMENT);                          // == [[T,Y,V,R],[Y,V,R,T]]
+ *
+ * // combinability
+ * auto v6 = vec | view::complement | view::translate_frames(WITH_REVERSE_COMPLEMENT);      // == [[C,M,H,A],[M,H,A,C]]
+ * ```
+ */
+seqan3::detail::translate_frames_fn const translate_frames;
+
+} // namespace seqan3::view
```

Comment:
You switch between namespaces quite often in this file, can you move stuff around so that you have a single `seqan3`, `seqan3::detail` and `seqan3::view` block (each)?

Reply:
I am switching namespaces because the implementation of the translate_frames_fn uses the view::translate_single which would not be defined if I would first put the detail namespace with both structs. Do you have an idea how that could work so that we can have only one detail and one view namespace?